### PR TITLE
Custom Pagination within Blog Section

### PR DESCRIPTION
**Pull request by AemieJ**
I replaced the template of the pagination of Hugo with custom pagination. In this custom pagination, it has the following properties: 

1. Show **<<<<** shortcut if not 1st page

page-start

2. Show **>>>>** shortcut if not the last page.

page-end

3. Show **<<** if there is a previous page
4. Show **>>** if there is the next page
5. Show current page number with the previous 2 pages and next 2 pages. Skip other pages with placeholder ...

page-middle

### CODE REVIEW

File: antora-ui-camel/src/css/doc.css [hunk 1]
```diff
@@ -368,7 +368,7 @@
 .doc ol,
 .doc ul {
   margin: 0;
-  padding: 0 0 0 2rem;
+  padding: 0 0 0 1rem;
```
Comment (zregvart) [inline antora-ui-camel/src/css/doc.css:371]: This change doesn't seem to be related to pagination. If it is the rule needs to be scoped to the pagination and placed in blog.css. Otherwise it affects other `ol` and `ul` elements.

File: layouts/blog/list.html [hunk 2]
```diff
@@ -15,9 +15,46 @@ <h3>Categories</h3>
             {{ range $pages }}
                 {{ .Render "summary" }}
             {{ end }}
-            {{ template "_internal/pagination.html" . }}
+            
+            {{ $page := .Paginator }}
+            {{ if gt $page.TotalPages 1 }}
+            {{ $.Scratch.Set "dot_rendered" false }}
+            <nav aria-label="page navigation">
```
Comment (zregvart) [inline layouts/blog/list.html:22]: Not sure we can use multiple strings for `aria-label`, [specification says](https://www.w3.org/TR/wai-aria/#aria-label) value in singular

File: layouts/blog/list.html [hunk 3]
```diff
@@ -15,9 +15,46 @@ <h3>Categories</h3>
...
+                <ul class="pagination">
+                    <!-- Don't show on 1st and 2nd page -->
```
Comment (zregvart) [inline layouts/blog/list.html:24]: Use [Hugo comments](https://gohugo.io/templates/introduction/#comments), no use of this in the rendered markup

File: layouts/blog/list.html [hunk 4]
```diff
@@ -15,9 +15,46 @@ <h3>Categories</h3>
...
+                    {{ if and (ne $page.PageNumber 1) }}
```
Comment (zregvart) [inline layouts/blog/list.html:25]: Perhaps

```suggestion
                    {{ if not .$page.First }}
```

File: layouts/blog/list.html [hunk 5]
```diff
@@ -15,9 +15,46 @@ <h3>Categories</h3>
...
+                    {{ end }}
+                </ul>
+            </nav>
+            {{ end }}
         </div>
     </main>
 </div>
 
-{{ partial "footer.html" . }}
+{{ partial "footer.html" . }}
```
Comment (zregvart) [inline layouts/blog/list.html:60]: No newline at EOF

File: layouts/blog/list.html [hunk 6]
```diff
@@ -15,9 +15,46 @@ <h3>Categories</h3>
...
+                    <li class="page-item"><a href="{{ $page.Next.URL }}" rel="next" class="page-link">»</a></li>
+                    {{ end }}
+
+                    <!-- Don't show on last and 2nd last page -->
+                    {{ if and (ne $page.PageNumber $page.TotalPages) }}
+                    <li class="page-item"><a href="{{ $page.Last.URL }}" rel="last" class="page-link">»»</a></li>
+                    {{ end }}
+                </ul>
+            </nav>
+            {{ end }}
```
Comment (zregvart) [inline layouts/blog/list.html:55]: 1. We're missing `aria-label` in `a` tags. Examine the [default template](https://github.com/gohugoio/hugo/blob/master/tpl/tplimpl/embedded/templates/pagination.html).
2. Use `.Last` as `.First` above
3. Avoid using scratch if possible, use explicit conditions or local variables instead (consult the internal Hugo template)

Comment (AemieJ) [reply layouts/blog/list.html:55]: Alright, I will make use of this and do the required changes.

File: layouts/blog/list.html (hunk 4 above)
Comment (AemieJ) [reply layouts/blog/list.html:25]: The command you suggested isn't performing. **not $page.First** doesn't provide any output.

File: layouts/blog/list.html (hunk 6 above)
Comment (AemieJ) [reply layouts/blog/list.html]: I have done except the 2nd one I had to use the **ne** condition as **not** wasn't given the expected results.

**Comment by zregvart** [comment]
Thanks!
